newIDE/app/src/Hints/index.js

```diff
@@ -1,8 +1,13 @@
 // @flow
 import { t } from '@lingui/macro';
 import { type MessageDescriptor } from '../Utils/i18n/MessageDescriptor.flow';
+import { type AlertMessageIdentifier } from '../MainFrame/Preferences/PreferencesContext';
 
-export type Hint = {| kind: 'warning' | 'info', message: MessageDescriptor |};
+export type Hint = {|
+  kind: 'warning' | 'info',
+  message: MessageDescriptor,
+  identifier?: AlertMessageIdentifier,
```

Comment:
Let's call this `dismissIdentifier` to make it very clear that if set, it's used to dismiss the alert.